addons/point_of_sale/models/pos_order.py

```diff
@@ -165,6 +169,10 @@ def _process_order(self, order, draft, existing_order):
 
         return pos_order.id
 
+    def _clean_payment_lines(self):
+        self.ensure_one()
```

Comment:
This new method is an extraction of code that was done in the process of orders (one by one), needed to be overriden by the new module to manage online payments.
Of course the method could be changed to handle several orders in self, but it is currently not useful. I prefer to don't add complexity to a codebase that is already complex (by complex I don't mean the change would be difficult, we can just replace the '=' by 'in' and self.id by self.ids, I mean complex in understanding).
I personally don't like the "implied by the attribute access" principle, because it means that anyone that wants to use that method needs to read the full implementation to check if the self can hold only one record or several. Writing "self.ensure_one()" at the beginning of the method avoids that research.